pymongo/collection.py

```diff
@@ -431,7 +431,7 @@ def with_options(
         read_preference: Optional[_ServerMode] = None,
         write_concern: Optional[WriteConcern] = None,
         read_concern: Optional[ReadConcern] = None,
-    ) -> Collection[_DocumentType]:
+    ) -> Collection[_DocumentTypeArg]:
```

Comment:
Could we revert this change and discuss it in a new jira ticket? I think the interaction of Collection and CodecOption's DocumentType is not well defined so I don't think we should make breaking changes like this without more consideration of the whole api.

Reply:
Yeah, I agree there isn't a good way to handle this with the typing system.  I think we should stick with what we've previously shipped.